common/short-hex-str/src/lib.rs

```diff
@@ -16,8 +16,12 @@ use thiserror::Error;
 pub struct ShortHexStr([u8; ShortHexStr::LENGTH]);
 
 #[derive(Error, Debug)]
-#[error("Input bytes are too short")]
-pub struct InputTooShortError;
+pub enum Error {
+    #[error("Input bytes are too short")]
+    InputTooShortError(),
```

Comment:
nit: this doesn't need empty brackets (as it takes no arguments), just defining the type should be enough, e.g.:

`InputTooShortError,`